### PR TITLE
fix: parquet column count statistics (#2124)

### PR DESCRIPTION
**Pull request by tustvold**
Initially StatValues tracked null_count and non-null count. At some point this was changed to track total_count and null_count. 

Unfortunately the parquet implementation doesn't appear to have been properly updated. This didn't cause any tests to fail as the parquet crate was internally consistent on treating `total_count` as the non-null count, including in its tests. This only caused issues when interfacing with other code in IOx.

In particular it causes log output such as

```
2022-01-11T19:18:55.861248Z  WARN data_types::partition_metadata: Mismatch in statistics count, see #2124 table_name=user_asset column_name=old_amount column_count=234152 previous_count=236852
```

When running IOx with data that would previously print this error, the warning is no longer printed.

Also related to https://github.com/influxdata/conductor/issues/763

### CODE REVIEW

File: parquet_file/src/metadata.rs [hunk 1]
```diff
@@ -715,8 +715,7 @@ fn read_statistics_from_parquet_row_group(
                 .fail()?;
             }
 
-            let count =
-                (row_group.num_rows().max(0) as u64).saturating_sub(parquet_stats.null_count());
+            let count = row_group.num_rows().max(0) as u64;
```
Comment (tustvold) [inline parquet_file/src/metadata.rs:718]: This is the change, the rest is just tests plumbing

File: parquet_file/src/metadata.rs [hunk 2]
```diff
@@ -952,7 +953,8 @@ mod tests {
         let n_rows = decoded.md.file_metadata().num_rows() as u64;
         assert!(n_rows >= decoded.md.num_row_groups() as u64);
         for summary in &chunk.table_summary().columns {
-            assert!(summary.total_count() <= n_rows);
+            assert_eq!(summary.total_count(), n_rows);
+            assert!(summary.null_count() <= summary.total_count());
```
Comment (alamb) [inline parquet_file/src/metadata.rs:957]: Could we also assert that `decoded.md.null_count() == summary.null_count()`?

**Comment by alamb** [comment]
Also possibly related https://github.com/influxdata/influxdb_iox/issues/2289 -- part of the same story

**Review by alamb — APPROVED** [review]
Nice catch

**Pull request merged**